presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1748,7 +1748,14 @@ public ConnectorPushdownFilterResult pushdownFilter(ConnectorSession session, Co
         //  - range filters that apply to subfields,
         //  - the rest
         ExtractionResult<Subfield> decomposedFilter = rowExpressionService.getDomainTranslator()
-                .fromPredicate(session, filter, new SubfieldExtractor(functionResolution, rowExpressionService.getExpressionOptimizer(), session).toColumnExtractor());
+                .fromPredicate(
+                        session,
+                        filter,
+                        new SubfieldExtractor(
+                                functionResolution,
+                                rowExpressionService.getExpressionOptimizer(),
+                                session,
+                                HiveSessionProperties.isNestedColumnsFilterEnabled(session)).toColumnExtractor());
```

Comment:
Just realized `toColumnExtractor` returns generic type. Could you fix the signature to be
```java
public DomainTranslator.ColumnExtractor<Subfield> toColumnExtractor()